### ISSUE TITLE
SETCLIP is broken

### ISSUE THREAD

**Issue by mr-martian** [apertium/apertium-recursive#53]
```xml
<let>
  <clip pos="1" part="gen" side="tl"/>
  <lit v="3"/>
</let>
```
If a rule gets applied multiple times, later invocations will see `<3>` and not the original tag.

**Comment by mr-martian** [comment]
The first `SETCLIP` that applies to an input node now makes a copy before editing, which solves the initially reported issue.

**Comment by ftyers** [comment]
Hmm, shouldn't they see 3, or should they see what `<3>` refers to ?

**Comment by mr-martian** [comment]
The issue here is that if you have a rule that matches `n` and outputs `NP` and it gets applied in multiple branches, the first application will get a target side with `<n><f><sg>` but in the second branch it will look like biltrans output `<n><2><3>`.

**Comment by ftyers** [comment]
Biltrans output would be `<n><GD><ND>` no ?

**Comment by mr-martian** [comment]
The actual biltrans output is `<n><GD><ND>`, but the effect of this bug is that rules sometimes see `<n><2><3>` instead.

I fixed the cases that I'm aware of in aef91f1, but I'm leaving this issue open because I think I might have missed some.
